src/js_udp_wrap.cc

```diff
@@ -100,10 +100,13 @@ ssize_t JSUDPWrap::Send(uv_buf_t* bufs,
     total_len += bufs[i].len;
   }
 
+  Local<Object> address;
+  if (!AddressToJS(env(), addr).ToLocal(&address)) return -1;
```

Comment:
-1 is `EPERM` – doesn’t really fit, and I don’t think we should be making up error codes on the go here. The default of `UV_EPROTO` is fine, I think, we also use that elsewhere in this situation.

Reply:
I used -1 here because of https://github.com/nodejs/node/pull/40425#discussion_r727264779. Do you think  we should change the error code there too?

Reply:
Yeah, I think that would be good actually 👍

Reply:
But then this is the same situation as the one in that PR, right? If the error code doesn't reach the JS caller because of the exception, perhaps it's better to use -1 (not the EPERM symbol because I don't want it to communicate an "operation not permitted" error)? The advantage is that it doesn't falsely convey that it's a real "protocol error" but just an unsuccessful case.

Reply:
That’s true for callers from JS, but these methods return a ssize_t rather than a JS return value because they are designed to be callable from other parts of our C++ codebase as well. (For example, TLS and HTTP/2 call stream methods on the underlying TCP socket this way.)

If clarity is the concern, then let’s maybe alias `using JS_EXCEPTION_PENDING = UV_EPROTO;` and stick with the former name across the board?

Reply:
Yes, I was concerned about clarity. `using JS_EXCEPTION_PENDING = UV_EPROTO;` looks like an interesting choice. Is there any reason for choosing `UV_EPROTO`? What do you think about using something like `1` which won't fall in the range of values covered by libuv errno values because it's a positive number?

Reply:
> What do you think about using something like 1 which won't fall in the range of values covered by libuv errno values because it's a positive number?

Well, the API contract here is that negative numbers represent errors and 0 or positive success, so I don’t think that will work :)

I think using another value would potentially be fine, but choosing a valid libuv error code was intentional here.

Reply:
Used `#define JS_EXCEPTION_PENDING UV_EPROTO` instead because `using JS_EXCEPTION_PENDING = UV_EPROTO;` doesn't compile since `UV_EPROTO` isn't a type.